tools/zetasql_helper/zetasql_helper/local_service/local_service.cc

```diff
@@ -33,5 +36,51 @@ absl::Status ZetaSqlHelperLocalServiceImpl::Tokenize(
   return absl::OkStatus();
 }
 
+absl::Status ZetaSqlHelperLocalServiceImpl::ExtractFunctionRange(
+    const ExtractFunctionRangeRequest *request,
+    ExtractFunctionRangeResponse *response) {
+
+  std::unique_ptr<::bigquery::utils::zetasql_helper::FunctionRange> output;
+  ZETASQL_RETURN_IF_ERROR(
+      ::bigquery::utils::zetasql_helper::ExtractFunctionRange(
+          request->query(), request->line_number(), request->column_number(), &output
+      ));
+
+  auto status_or_function_range_proto = output->ToProto();
+  ZETASQL_RETURN_IF_ERROR(status_or_function_range_proto.status());
```

Comment:
why return status separately? if error occurs, should it return at line 44? or there could be cases line 44 is ok but getting status() is not ok?

Reply:
Yes. Line 44 makes sure the ExtractFunctionRange is correct. Then we get a FunctionRange object. Then we need to serialize it to proto, which is line 50. Here may  occur errors.